src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Gets the processor for the specified rule type.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\DateTimeProvider\CurrentDateTimeProvider;
+use \Automattic\WooCommerce\Admin\PluginsProvider\PluginsProvider;
+
+/**
+ * Class encapsulating getting the processor for a given rule type.
+ */
+class GetRuleProcessor {
```

Comment:
Probably too big of a change at this point, but it could be interesting to consider a filter driven approach... so we could just do something like...

```
function wc_remote_inbox_process_plugins_activated_rule( $result, $rule ) {
    // process
}
add_filter( 'wc_remote_inbox_process_plugins_activated_rule', 'wc_remote_inbox_process_plugins_activated_rule', 10, 3 );

apply_filters( 'wc_remote_inbox_process_plugins_activated_rule', $result, $rule );
```

That would allow for additional extensibility and easy hooking into things for logging/metrics/testing/etc.